lms/djangoapps/discussion/rest_api/views.py

```diff
@@ -578,6 +587,95 @@ def partial_update(self, request, comment_id):
         return Response(update_comment(request, comment_id, request.data))
 
 
+class CanPostOrComment(permissions.BasePermission):
```

Comment:
I'd move this method to `permissions.py` 

You can name with respect to DRF naming conventions. e.g. `IsAuthticated` . Let's name it `IsStaffOrCourseTeamOrEnrolled`.